Source/WTF/wtf/HashTable.h

```diff
@@ -886,12 +879,14 @@ DECLARE_ALLOCATOR_WITH_HEAP_IDENTIFIER(HashTable);
             entry = table + i;
             
             // We count on the compiler to optimize out this branch.
-            if constexpr (HashFunctions::safeToCompareToEmptyOrDeleted) {
+            if constexpr (HashFunctions::safeToCompareToEmptyOrDeleted && shouldValidateKey != ShouldValidateKey::Yes) {
                 if (isEmptyBucket(*entry))
                     break;
                 
-                if (HashTranslator::equal(Extractor::extract(*entry), key))
+                if (HashTranslator::equal(Extractor::extract(*entry), key)) {
+                    validateKey<ShouldValidateKey::No>(*entry);
```

Comment:
I think it would be clearer to have a non-template function to use in a case like this. It’s so wordy to pass the template argument.